code/modules/mob/living/silicon/robot/robot_defense.dm

```diff
@@ -320,9 +320,19 @@ GLOBAL_LIST_INIT(blacklisted_borg_hats, typecacheof(list( //Hats that don't real
 		return
 	switch(severity)
 		if(1)
-			Stun(160)
+			emp_knockout(16 SECONDS)
 		if(2)
-			Stun(60)
+			emp_knockout(6 SECONDS)
+
+/mob/living/silicon/robot/proc/emp_knockout(deciseconds)
+	if(!src || !deciseconds)
```

Comment:
Is there a good reason to assume that `src` might not exist here? Can you be destroyed in `emp_act` before reaching this code?
If so you should probably just check that in `emp_act` instead.

```
. = ..()
if (. & EMP_PROTECT_SELF || QDELETED(src))
	return
```

Reply:
Sure, but if there's any reason to suspect that it might happen you should check it as early as possible and not bother doing any other unecessary work, so please check it via `QDELETED(src)` in `emp_act` instead of in this proc.